privacyidea/lib/token.py

```diff
@@ -2160,12 +2160,14 @@ def check_token_list(tokenobject_list, passw, user=None, options=None, allow_res
         message_list = ["matching {0:d} tokens".format(len(valid_token_list))]
         # write serial numbers or something to audit log
         for token_obj in valid_token_list:
-            if increase_auth_counters:
-                token_obj.inc_count_auth_success()
-            # Reset the failcounter, if there is a timeout set
-            token_obj.check_reset_failcount()
-            # Check if the max auth is succeeded
+            # Check if the max auth is succeeded.
+            # We need to set the offsets, since we are in the n+1st authentication.
             if token_obj.check_all(message_list):
+                if increase_auth_counters:
+                    token_obj.inc_count_auth_success()
+                # Reset the failcounter, if there is a timeout set
+                token_obj.check_reset_failcount()
```

Comment:
Sorry I'm only noticing this now, but I think this change breaks the ``failcounter_clear_timeout`` feature.
Assume we set ``failcounter_clear_timeout`` to 3 minutes and there is a token whose failcounter is exceeded.
After 5 minutes, we try to authenticate with a correct OTP.
We should be able to authenticate because the last unsuccessful authentication was >3 minutes ago.
But because the fail counter is exceeded, ``check_all`` returns False, so we cannot authenticate.

I think we have to move ``token_obj.check_reset_failcount()`` before ``check_all()`` again?

Reply:
Thanks a lot for pointing this out.
But this is strange, since I *think* we have test cases for the resetting of the failcounter.

I will check this, add a testcase and adapt this if necessary!